### ISSUE TITLE
GlideN64 default framebuffer

### ISSUE THREAD

**Issue by loganmc10** [Provenance-Emu/Provenance#780]
I saw that this was merged: https://github.com/gonetz/GLideN64/commit/ff904fc51ef2476600f18c088e8a0b0616752057

IOS isn't the only platform that has this issue (0 isn't the right framebuffer to render into). I'd like to extend the mupen64plus API to support this situation. Do you use the VidExt API for mupen64plus? (https://github.com/mupen64plus/mupen64plus-core/blob/master/doc/emuwiki-api-doc/Mupen64Plus-v2.0-Core-Video-Extension.mediawiki)

If so, I would propose adding a new function to the API (VidExt_GetDefaultFramebuffer), and the GFX plugin could just query that. Would you be open to this solution? It would allow GLideN64 to support other platforms with the same issue without a bunch of ifdefs

**Comment by braindx** [comment]
Yes, we do use the VidExt API and that seems like a sensible solution.  One note: it should probably be named VidExt_**GL**_GetDefaultFramebuffer since it's OpenGL specific.

**Comment by loganmc10** [comment]
Yes good point, ok well we'll leave it like it is now, sometime next week I'll submit a PR to mupen64plus-core

**Comment by JoeMatt** [comment]
We need to update our mupen64plus. I have to submodule it like we did with GLiden64 since it's a pain to pull / push upstream right now. I'll take a stab at it, requires basically recreating our Xcode project, finding the correct files to include etc. We could try using the cmake apart of our build process to avoid that though that's one more thing to fail for users who aren't savvy in Xcode and are just installing / using it to side load our app.

**Comment by loganmc10** [comment]
Here is the PR: https://github.com/mupen64plus/mupen64plus-core/pull/556

**Comment by JoeMatt** [comment]
Nice, I'll get on updating our mupen64plus core. Hopefully it just works. We're on OpenEMU's fork at the moment which is pretty old and stuff as moved around so I have to hand roll a new Xcode project or incorporate Make into our build and link the target lib into the framework (pros and cons there, maintainability vs possible errors/difficulty for other users compiling due to our side-loading nature requiring non-devs to work with Xcode for possibly the first time). Xcode does have native support for Make and CMake but it externs it out to shell calls and can be difficult to see / debug build errors. We might have to tweak the Makefile too, I'll have to experiment with it.

**Comment by loganmc10** [comment]
Just as a heads up, here is the change I plan on submitting to GLideN64 once the mupen64plus-core change is merged:

https://github.com/gonetz/GLideN64/compare/master...loganmc10:vidext

**Comment by loganmc10** [comment]
Ok the changes have been merged into mupen64plus, and I submitted a PR to GLideN64 here: https://github.com/gonetz/GLideN64/pull/1810

So basically you just need to update mupen64plus-core, implement the VidExt_GL_GetDefaultFramebuffer function in your VidExt implementation, and test out that GLideN64 PR
